src/supplemental/nanolib/parquet/parquet.cc

```diff
@@ -688,11 +702,10 @@ parquet_file_queue_init(conf_parquet *conf)
 						         "file count: %d.",
 						    conf->file_count);
 					} else {
+						free(file_path);
```

Comment:
This could result in a double free situation.

Reply:
don't understand, file_path not enqueue parquet_file_queue, where will cause double free?

Reply:
See function remove_old_file(void).

Reply:
But this file_path was created by alloc and is not queued yet. The file_path in remove_old_file() is taken from the queue.